app/questionnaire/questionnaire_schema.py

```diff
@@ -248,6 +248,15 @@ def get_relationship_collectors_by_list_name(self, list_name: str):
                 if block["for_list"] == list_name
             ]
 
+    def get_list_name(self, current_location):
+        list_name = None
+        repeating_list = self.get_repeating_list_for_section(
+            current_location.section_id
+        )
+        if repeating_list and current_location.list_item_id:
+            return repeating_list
+        return list_name
```

Comment:
Could remove the `list_name = None` and just `return None` if you want to be explicit, ot just removed entirely.